cairo/ethereum/cancun/transactions.cairo

```diff
@@ -215,3 +224,166 @@ func signing_hash_4844{range_check_ptr, bitwise_ptr: BitwiseBuiltin*, keccak_ptr
     let hash = keccak256(encoded_tx);
     return hash;
 }
+
+func recover_sender{
+    range_check_ptr,
+    range_check96_ptr: felt*,
+    add_mod_ptr: ModBuiltin*,
+    mul_mod_ptr: ModBuiltin*,
+    bitwise_ptr: BitwiseBuiltin*,
+    keccak_ptr: KeccakBuiltin*,
+    poseidon_ptr: PoseidonBuiltin*,
+}(chain_id: U64, tx: Transaction) -> (Address, EthereumException*) {
+    alloc_locals;
+    tempvar SECP256K1N = U256(new U256Struct(low=secp256k1.N_LOW_128, high=secp256k1.N_HIGH_128));
+    tempvar SECP256K1N_DIVIDED_BY_2 = U256(
+        new U256Struct(low=secp256k1.N_DIVIDED_BY_2_LOW_128, high=secp256k1.N_DIVIDED_BY_2_HIGH_128)
+    );
+    tempvar zero = U256(new U256Struct(low=0, high=0));
+
+    if (cast(tx.value.legacy_transaction.value, felt) != 0) {
+        let r = tx.value.legacy_transaction.value.r;
+        let s = tx.value.legacy_transaction.value.s;
```

Comment:
I think it would be acceptable to do a TxImpl, in which you have a function `get_signature_components` that returns (r, s, v) supporting all transaction types. 

Once this is done you can move out the r, s checks step (which is copy/pasted in each tx variant) to be done right before. it's a very small overhead to pay for simplicity. Open to discuss

Reply:
I'd vote to do this only if another function requires r, s, v to be extracted from Transaction

Reply:
ok i'll do some utils for check_transaction pr